dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -15,21 +15,28 @@ sealed class TypedColumn[T, U](
 )(
   implicit
   val uencoder: TypedEncoder[U]
-) extends UntypedExpression[T] {
+) extends UntypedExpression[T] { self =>
 
   def this(column: Column)(implicit uencoder: TypedEncoder[U]) {
     this(FramelessInternals.expr(column))
   }
 
   def untyped: Column = new Column(expr)
 
-  def ===(other: U): TypedColumn[T, Boolean] = {
-    new TypedColumn[T, Boolean](untyped === other)
-  }
+  def ===(other: U): TypedColumn[T, Boolean] = new TypedColumn[T, Boolean](self.untyped === other)
+  def ===(other: TypedColumn[T, U]): TypedColumn[T, Boolean] = new TypedColumn[T, Boolean](self.untyped === other.untyped)
 
-  def ===(other: TypedColumn[T, U]): TypedColumn[T, Boolean] = {
-    new TypedColumn[T, Boolean](untyped === other.untyped)
-  }
+  def +(u: TypedColumn[T, U])(implicit n: Numeric[U]):  TypedColumn[T, U]= new TypedColumn[T, U](self.untyped.plus(u.untyped))
```

Comment:
I agree. I think we did something similar for "Ordering" and it makes perfect sense to have our types here. This was a quick first cut to get some early feedback. I am very happy you like it. I will try to polish it it up.